pkg/controller/cloneset/update/cloneset_update.go

```diff
@@ -141,8 +141,9 @@ func calculateUpdateCount(coreControl clonesetcore.Control, strategy appsv1alpha
 	}
 	waitUpdateIndexes = waitUpdateIndexes[:(len(waitUpdateIndexes) - partition)]
 
+	roundUp := !(strategy.MaxSurge != nil && strategy.MaxSurge.IntValue() > 0)
```

Comment:
The question is, if `maxSurge` is percentage str, `maxSurge.IntValue()` will also return 0.

How about this:

```golang
roundUp := true
if strategy.MaxSurge != nil {
    roundUp = strategy.MaxSurge.Type == intstr.Int && strategy.MaxSurge.IntValue() == 0
}
```

Reply:
But it cannot cover string type case.

Reply:
how about this:
` 
roundUp := true
if strategy.MaxSurge != nil {
   maxSurge, _ := intstrutil.GetValueFromIntOrPercent(strategy.MaxSurge, totalReplicas, true)
   roundUp = maxSurge == 0
}

`

Reply:
@FillZpp

Reply:
@shiyan2016 That's ok